addons/web/static/src/views/fields/properties/property_definition_selection.js

```diff
@@ -250,16 +238,39 @@ export class PropertyDefinitionSelection extends Component {
             destinationOptionIndex++;
         }
 
-        // we might be editing the value and move it immediately after
-        options[movedOptionIndex][1] = this.propertyDefinitionSelectionRef.el.querySelector(
-            `div[option-name="${movedOption}"] input`
-        ).value;
+        const activeEl = document.activeElement;
+        if (
+            activeEl &&
+            this.propertyDefinitionSelectionRef.el.contains(activeEl) &&
+            activeEl.tagName === "INPUT"
+        ) {
+            const optionName = activeEl
+                .closest(".o_field_property_selection_option")
+                .getAttribute("option-name");
+            const editedOptionIndex = options.findIndex((option) => option[0] === optionName);
+            // we might be editing the value and drag and drop something else just after
```

Comment:
If I understood correctly, this is more:

"As we cancel the blur event (to allow moving an empty option without deleting it), we need to manually update the value inside the options to save what the user may have input into it right before moving it around".

Reply:
noy only, you can also edit a "non new option" and then move an other one

without this line, the edition won't be saved